exporter/splunkhecexporter/client.go

```diff
@@ -148,16 +263,46 @@ func (c *client) sendSplunkEvents(ctx context.Context, splunkEvents []*splunk.Ev
 	return nil
 }
 
-func (c *client) pushLogData(ctx context.Context, ld pdata.Logs) error {
-	c.wg.Add(1)
-	defer c.wg.Done()
+// subLogs returns a subset of `ld` starting from index `start` to the end.
+func subLogs(ld *pdata.Logs, start *logIndex) *pdata.Logs {
+	if ld == nil || start == nil || start.zero() {
+		return ld
+	}
 
-	splunkEvents := logDataToSplunk(c.logger, ld, c.config)
-	if len(splunkEvents) == 0 {
-		return nil
+	logs := pdata.NewLogs()
+
+	rl, rl2 := ld.ResourceLogs(), logs.ResourceLogs()
+
+	for r := start.resource; r < rl.Len(); r++ {
+		rl2.Append(pdata.NewResourceLogs())
+		rl.At(r).Resource().CopyTo(rl2.At(r - start.resource).Resource())
+
+		il, il2 := rl.At(r).InstrumentationLibraryLogs(), rl2.At(r-start.resource).InstrumentationLibraryLogs()
+
+		i := 0
+		if r == start.resource {
+			i = start.library
+		}
+		for i2 := 0; i < il.Len(); i++ {
+			il2.Append(pdata.NewInstrumentationLibraryLogs())
+			il.At(i).InstrumentationLibrary().CopyTo(il2.At(i2).InstrumentationLibrary())
+
+			lr, lr2 := il.At(i).Logs(), il2.At(i2).Logs()
```

Comment:
Nit: I think the usage of usage of multiple assignment in this situation has no benefit and results in an increased cognitive load when reading this line, compared to 2 lines of regular assignments.